code/modules/mob/living/carbon/xenomorph/Facehuggers.dm

```diff
@@ -219,35 +231,88 @@
 		if(!attached) //Didn't hit anything?
 			i = 5
 			for(M in loc)
-				if(!i) break
+				if(!i)
+					break
 				if(CanHug(M))
 					Attach(M)
 					break
 				i--
 
-/obj/item/clothing/mask/facehugger/proc/Attach(mob/living/M)
-	set waitfor = 0
+/obj/item/clothing/mask/facehugger/throw_at(atom/target, range, speed)
+	. = ..()
+	if(stat == CONSCIOUS)
+		update_stat(UNCONSCIOUS) //stopping their process for the flight duration.
 
-	if(attached || M.status_flags & XENO_HOST || isXeno(M) || loc == M || stat != CONSCIOUS)
-		return
+/obj/item/clothing/mask/facehugger/throw_impact(atom/hit_atom, speed)
+	if(ismob(hit_atom) && stat != DEAD)
```

Comment:
Add the `if(stat == DEAD)` as an early return, not to mess with the `else` below.